zero_bin/rpc/src/main.rs

```diff
@@ -56,6 +57,15 @@ impl Cli {
             } => {
                 let checkpoint_block_number =
                     checkpoint_block_number.unwrap_or((start_block - 1).into());
+                if let BlockId::Number(checkpoint_block_number) = checkpoint_block_number {
```

Comment:
I see now that `checkpoint_block_number` in the leader is `u64`. You could change `checkpoint_block_number` rpc cli parameter type from `BlockId` to `u64`, then no need for all this de-structuring `if/else`